odoo/models.py

```diff
@@ -1968,10 +1970,8 @@ def _read_group(self, domain, groupby=(), aggregates=(), having=(), offset=0, li
         ]
         if query.where_clause:
             query_parts.append(SQL("WHERE %s", query.where_clause))
-        if groupby_terms:
-            if sql_extra_groupby:
-                groupby_terms.append(sql_extra_groupby)
-            query_parts.append(SQL("GROUP BY %s", SQL(", ").join(groupby_terms)))
+        if query.groupby:
```

Comment:
You could just add `having` to `Query` and we would not need anymore the split into query parts.